psi4/driver/driver.py

```diff
@@ -106,6 +108,8 @@ def _find_derivative_type(ptype, method_name, user_dertype):
         raise ValidationError("""Derivative method 'name' %s and derivative level 'dertype' %s are not available.%s"""
                               % (method_name, str(dertype), alternatives))
 
+    dertype = min(dertype, derivatives[ptype])
```

Comment:
this is in case Hess available but only grad needed?

I can believe this fn has some edge cases. I ended up refactoring it https://github.com/psi4/psi4/pull/1351/files#diff-a7c88e038c7c4c8ab84033e68c2b7c38R227

Reply:
Right. After the first commit, I realized there was an edge case where we were failing: if `dertype=2` for a method with hessians (currently RHF), the gradient code would see that `dertype!=1` and default to getting the gradient by energies, which is rather silly. Hence the second commit, where I added this line.

What do you recommend I do - proceed as-is but realize this is a short-term fix until #1351, or revert the second commit? Only the first commit is essential.

Reply:
This is good for now. #1570 only adds the fn -- it won't be hooked up until #1351. just be aware that you'll have to test again that the path you want works under new scheme. there's a test_dertype can use to confirm.